sentinelhub/api/batch/statistical.py

```diff
@@ -0,0 +1,215 @@
+"""
+Module implementing an interface with
+`Sentinel Hub Batch Processing API <https://docs.sentinel-hub.com/api/latest/api/batch-statistical/>`__.
+"""
+import datetime as dt
+import logging
+import sys
+from dataclasses import dataclass, field
+from typing import Any, List, Optional, Union
+
+from dataclasses_json import CatchAll, LetterCase, Undefined
+from dataclasses_json import config as dataclass_config
+from dataclasses_json import dataclass_json
+
+from ...type_utils import Json, JsonDict
+from ..base_request import InputDataDict
+from ..statistical import SentinelHubStatistical
+from ..utils import datetime_config, enum_config, remove_undefined
+from .base import BaseBatchClient, BaseBatchRequest, BatchRequestStatus, BatchUserAction
+
+if sys.version_info < (3, 11):
+    from typing_extensions import NotRequired, TypedDict
+else:
+    from typing import NotRequired, TypedDict  # pylint: disable=ungrouped-imports
+
+LOGGER = logging.getLogger(__name__)
+
+BatchStatisticalRequestType = Union[str, dict, "BatchStatisticalRequest"]
+
+
+class S3AccessSpecification(TypedDict):
+    """Specification of a S3 path."""
+
+    url: str
+    accessKey: str
+    secretAccessKey: str
+    region: NotRequired[str]
+
+
+class S3Specification(TypedDict):
+    """Specification of a S3 input or output."""
+
+    s3: S3AccessSpecification
+
+
+class SentinelHubBatchStatistical(BaseBatchClient["BatchStatisticalRequest"]):
+    """An interface class for Sentinel Hub Batch Statistical API
+
+    Check `Batch Statistical API <https://docs.sentinel-hub.com/api/latest/reference/#tag/batch_statistical>`__ for more
+    information.
+    """
+
+    @staticmethod
+    def _get_service_url(base_url: str) -> str:
+        """Provides URL to Batch Statistical API"""
+        return f"{base_url}/api/v1/statistics/batch"
+
+    def create(
+        self,
+        input_features: S3Specification,
+        input_data: List[Union[JsonDict, InputDataDict]],
+        aggregation: JsonDict,
+        calculations: JsonDict,
+        output: S3Specification,
+        **kwargs: Any,
+    ) -> "BatchStatisticalRequest":
```

Comment:
If you used

```Python
from __future__ import annotations
```
the return type here could be `BatchStatisticalRequest` instead of `"BatchStatisticalRequest"`, right? Although I'm not sure if that would make it in any way better than now.

(Same comment applies fro the method below.)